dev/com.ibm.ws.ejbcontainer.core/test/com/ibm/ejs/container/ScheduleExpressionParserTest.java

```diff
@@ -2088,53 +1994,226 @@ public void testStartAndEnd()
     }
 
     @Test
-    public void testDaylightSavingsTime()
-    {
+    public void testDaylightSavingsTime() {
         // This test relies on US-specific DST transitions in 2009.
+
+        // First, cover springing ahead to DST
```

Comment:
Can we get a similar fall back test where the hour isn't a wild card?

I'm not sure if I'm convinced ParsedScheduleExpression.getTimeout() doesn't need anything in the `if (hours != WILD_CARD) {` check on line 589

Reply:
like if the scheduled expression is specifically hour = 2, wildcard the rest.

Reply:
I've added scenarios for hour("2"), hour("2-4"), hour("2).minute("*"), and hour("2-4").minute("*").second("*")

Not adding any code to `if (hours != WILD_CARD) {` check on line 589 allowed a couple of subtle problems which probably wouldn't be noticed, but are:
1 - For spring DST, if the schedule is "2" only, then the next timeout would advance to "2" two days later, rather than just the next day
2 - For fall DST, the first 1 AM would be skipped if minute and second are both "*".

Code has been added to correct both of thse.